.travis.yml

```diff
@@ -23,32 +23,29 @@ services:
 git:
   submodules: false
 
+# This script is triggered only with python2.7 (as below python3.6 is excluded from the matrix)
 before_install:
   - sudo apt update && sudo apt install -y protobuf-compiler
   - sed -i 's,git\@github.com:\([^/]*\)/\(.*\).git,https://github.com/\1/\2,' .gitmodules
   - git submodule update --init --recursive
 
 before_script:
   - pip install -r source/tyr/requirements_dev.txt
-  - pip install pre-commit
-  - pre-commit install
   - bash source/scripts/build_protobuf.sh
 
 script:
   - bash source/scripts/check_submodules.sh
-  - pre-commit run clang-format-6.0 --all --show-diff-on-failure
   - pushd source/tyr && PYTHONPATH=.:../navitiacommon/ py.test --doctest-modules --ignore=migrations/ ; popd
-  - pre-commit run flake8 --all
 
 matrix:
-  include:
+  include:  # This script is triggered only with python3.6
     - python: "3.6"
       before_script:
-        - python3.6 -m pip install black==18.9b0
-        - pip install -r source/jormungandr/requirements_dev.txt
+        - pip install -r requirements_pre-commit.txt
+        - pip install pre-commit
+        - pre-commit install
         - bash source/scripts/build_protobuf.sh
       script:
-        - black --check --diff .
-        - pushd source/jormungandr && MYPYPATH=.:../navitiacommon mypy --ignore-missing-imports --py2 jormungandr && popd
+        - pre-commit run --all --show-diff-on-failure
```

Comment:
Does this mean that all the hooks in the pre-commit file will be run only for python 3?

Reply:
When nothing is specified in the pre-commit-config, yes.

But actually for flake8 it uses python2.7 that's available when launching pre-commit here (if launched with python3.6, flake8 currently fails).

That's the way I launch it locally (default = 3.6), I don't know if we may end with different behaviors on hooks without version specified. But then we will just pin them to one of the version, as it's working for both flake8(2.7) and black(3.6).